File: docs/src/testers/scraping/data_collection.py
# @sniptest filename=data_collection.py
from notte_sdk import NotteClient
from pydantic import BaseModel


class ProductInfo(BaseModel):
    name: str
    price: float
    rating: float | None
    reviews_count: int | None


client = NotteClient()

urls = [
    "https://store.example.com/product/1",
    "https://store.example.com/product/2",
]

products: list[ProductInfo] = []
for url in urls:
    data = client.scrape(url, response_format=ProductInfo)
    products.append(data)

File: docs/src/testers/scraping/data_collection.ts
import { z } from 'zod';
import { NotteClient } from 'notte-sdk';

const ProductInfo = z.object({
  name: z.string(),
  price: z.number(),
  rating: z.number().nullable(),
  reviews_count: z.number().nullable(),
});

const client = new NotteClient({
  apiKey: process.env.NOTTE_API_KEY,
});

const urls = [
  'https://store.example.com/product/1',
  'https://store.example.com/product/2',
];

const products = [];
for (const url of urls) {
  const data = await client.scrape(url, {
    response_format: ProductInfo,
  });
  products.push(data);
}
